src/sentry/search/snuba/executors.py

```diff
@@ -433,6 +419,26 @@ def query(
                     result_group_ids.add(group_id)
                     result_groups.append((group_id, group_score))
 
+            if hits is None:
```

Comment:
I don't get why this was moved here. 
By doing so you removed an optimization that allowed us to returnand empty set if hit count was 0 at line 362 without having to do the full snuba query. Are you planning to refactor the query method in such a way that we cannot calculate hits where they were calculated before ?

Reply:
Aaah you're right.

I was attempting to consolidate hits calculation's in one place, but I lost the ability to return an empty result set before ever running the full snuba query.

I fixed this - hits is now assigned in two places (potentially), but that optimization is back in.